rfcs/062-content-api-all-search/README.md

```diff
@@ -0,0 +1,132 @@
+
+## "All" index
+
+We will be creating a single index in Elasticsearch containing all Addressable content types in their most minimalistic form (**TODO add link detailed this**).
+
+### Addressable content types
+
+Here is a list of which Prismic content types we consider to be Addressable, in that their documents are all accessible to our users under a UID-based URL.
+
+This list also link to a file which describes what they are to look like in the Elasticsearch index. You may consult [the complete list here](./transformed-documents) instead.
+
+- **Events**: [Transformed indexed Event example](./transformed-documents/eventDocument.ts)
+- **Exhibitions**: [Transformed indexed Exhibition example](./transformed-documents/exhibitionDocument.ts)
+- **Stories**: [Transformed indexed Story example](./transformed-documents/storyDocument.ts)
+- **Pages**: [Transformed indexed Page example](./transformed-documents/pageDocument.ts)
+- **Visual stories**: [Transformed indexed Visual story example](./transformed-documents/visualStoryDocument.ts)
+- **Exhibition text**: [Transformed indexed Exhibition Text example](./transformed-documents/exhibitionTextDocument.ts)
+- **Exhibition highlight tour**: [Transformed indexed Exhibition Highlight examples](./transformed-documents/exhibitionHighlightDocument.ts). This document gets transformed into two different ones, consult [Exhibition highlight tour section](#exhibition-highlight-tour) below.
+- **Books**: [Transformed indexed Book example](./transformed-documents/bookDocument.ts)
+- **Projects**: [Transformed indexed Project example](./transformed-documents/projectDocument.ts)
+- **Seasons**: [Transformed indexed Season example](./transformed-documents/seasonDocument.ts)
+
+### Exhibition highlight tour
+
+This document is a special case, in that it is one Prismic document that needs to be indexed as two documents: "Audio with transcripts" and "British sign language with subtitles", as they are two different pages on the website ([Audio with transcripts](https://wellcomecollection.org/guides/exhibitions/jason-and-the-adventure-of-254/audio-without-descriptions) and [British sign language with subtitles](https://wellcomecollection.org/guides/exhibitions/jason-and-the-adventure-of-254/bsl)).
+
+### Have all fields in the "query" object align
+
+Something that will help the search performance would be to have as little fields to look through as possible, and have their names match across content types. We still want certain fields to have more weight than others in terms of relevance, so an option would be to list all the Prismic fields we want to have for each content type, and order them in terms of importance. Then, in the index, the query object could look like:
+
+```
+query: {
+    weight100: "[title]. [subtitle].",
```

Comment:
This pattern of naming indexed query fields by weight is not something we seem to do elsewhere and might not be slicing things in a useful way. I think it makes more sense to give these names related to their content and leave any reference of weighting in search to the search query.

Reply:
Works for me, so it's just about making sure that all 10 content types are using the same term for what serves the same purpose (like mentioned in the Captions, standfirsts and intro texts" of this document) then.

Reply:
> so it's just about making sure that all 10 content types are using the same term for what serves the same purpose

👍

Reply:
If we do what we said and add the strings of other fields in "description", maybe. Or add a fourth one ("contributors") as I feel like that's the main outlier that should have influence (e.g. if you search by artist name)

Reply:
[Now addressed](https://github.com/wellcomecollection/docs/pull/112/commits/a2f2201a4e59530214fce9b9b5cdcad731f238da)